src/wallet/wallet.cpp

```diff
@@ -310,7 +312,18 @@ std::shared_ptr<CWallet> CreateWallet(interfaces::Chain& chain, interfaces::Coin
             return nullptr;
         }
         if (!create_blank) {
-            {
+            // Unlock the wallet
+            if (!wallet->Unlock(passphrase)) {
+                error = Untranslated("Error: Wallet was encrypted but could not be unlocked");
+                status = DatabaseStatus::FAILED_ENCRYPT;
+                return nullptr;
+            }
+
+            // Set a seed for the wallet
+            if (wallet->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS)) {
+                LOCK(wallet->cs_wallet);
+                wallet->SetupDescriptorScriptPubKeyMans();
+            } else {
```

Comment:
Is my understanding correct that descriptor wallets will be sans HD?

Reply:
Note: HD wallets have deterministic way to calculate private keys from a seed, but non-HD wallets have completely random keys which makes difficult to backup them or to use as hardware wallets.

**Descriptors wallets are descriptor hierarchical deterministic wallets (HD wallets).**

The difference between legacy wallets (current one) and descriptor wallets is how they keep private keys and scripts.

Legacy wallets creates a bunch of private keys (even HD wallets still keep 1000 of keys and an information how they are derived to be sure that same private key won't be re-used due to re-using same derivation). 

Descriptor wallets instead keep a ranges, such as (0, 1000) + derivation path to use.

There's one more thing is different which is important in terms of implementation but not useful at the moment:
descriptor wallets have that many different types of descriptor how many types of addresses wallet support. For Bitcoin descriptor wallets have 3 types of descriptors: `pkh()/44`, `sh(wpkh())/49`, `wpkh()/84'` for Dash is only `pkh()/44'`. So, legacy wallets keep only private keys and no information which type of address should be used, for descriptor wallets descriptor enforces to use one and only one type of address. that's not important at the moment, but if we would introduce any new type of address (such as Shnor/taproot addresses), our descriptor wallets will have extra type of descriptor then.

Reply:
Thanks for clarifying, great to know it's using BIP44! I was concerned that it might use the old derivation system by default (having a consistent one is especially important when that `migratewallet` RPC come over since it'd be odd if legacy wallets migrated have a BIP44 derivation but fresh descriptor wallets have a BIP32 derivation) but that's been put to rest.

One small change needed, the derivation path needs to be `44'/5'/0'/0`, not `44'/0'/0'/0` (the second number is the coin discriminator and Dash has been assigned the value `5`, see [SatoshiLabs registration](https://github.com/satoshilabs/slips/blame/f246c8b4861ecfb53f0741904bab4fc624a9fb4c/slip-0044.md#L36)). This probably got skipped over since all testnets have been assigned `1`.

**Descriptor wallets:**

![HD descriptor wallets with Bitcoin's derivation path](https://github.com/dashpay/dash/assets/63189531/f1617958-3f43-42e0-87ca-9d28de49b275)

**Legacy wallets:**

![getwalletinfo output for legacy wallet](https://github.com/dashpay/dash/assets/63189531/3bcf30d1-9092-42eb-9a3e-f63bd7c64a3d)

![dumpwallet output for legacy wallet](https://github.com/dashpay/dash/assets/63189531/d514f4c8-43b4-4238-a433-3943419d7753)

---

The value can be fetched from `Params().ExtCoinType()`

Reply:
good catch @kwvg! Thanks for paying attention to derivation path. Fixed in the latest push